fasttrips/PathSet.py

```diff
@@ -888,6 +899,27 @@ def calculate_cost(STOCH_DISPERSION, pathset_paths_df, pathset_links_df, veh_tri
                            (cost_accegr_df[Passenger.PF_COL_LINK_MODE]             == PathSet.STATE_MODE_EGRESS)& \
                            (cost_accegr_df[Passenger.TRIP_LIST_COLUMN_TIME_TARGET] == 'departure'), "var_value"] = 0.0
 
+        # depart before preferred or arrive after preferred means the passenger just missed something important
+        cost_accegr_df.loc[(cost_accegr_df[PathSet.WEIGHTS_COLUMN_WEIGHT_NAME]     == "pat_variance"    )& \
+                           (cost_accegr_df[Passenger.PF_COL_LINK_MODE]             == PathSet.STATE_MODE_ACCESS)& \
+                           (cost_accegr_df[Passenger.TRIP_LIST_COLUMN_TIME_TARGET] == 'arrival'), "var_value"] = 0.0
+        cost_accegr_df.loc[(cost_accegr_df[PathSet.WEIGHTS_COLUMN_WEIGHT_NAME]     == "pat_variance"    )& \
+                           (cost_accegr_df[Passenger.PF_COL_LINK_MODE]             == PathSet.STATE_MODE_EGRESS)& \
+                           (cost_accegr_df[Passenger.TRIP_LIST_COLUMN_TIME_TARGET] == 'arrival'), "var_value"] = \
+            (cost_accegr_df[Passenger.PF_COL_PAX_B_TIME] - cost_accegr_df[Passenger.TRIP_LIST_COLUMN_ARRIVAL_TIME])/np.timedelta64(1,'m')
```

Comment:
This might need a little more explaining [ for me ] of the logic and how it meshes with the statements above.

Reply:
This is reworked with commit 13c47b7. I also added verbose commenting and a couple of assert statements to help with this. 

This is one of the most challenging sections of the calculate_cost() method to understand. I spend a long time here early in the project debugging and stepping through code to get a clear understanding.